docs/management.yml

```diff
@@ -19,267 +19,241 @@ tags:
-          description: Unauthorized.
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/responses/UnauthorizedError'
+          $ref: '#/components/responses/UnauthorizedError'
         403:
-          description: Forbidden.
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/responses/ForbiddenError'
-        404:
-          description: Not Found.
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/responses/NotFoundError'
+          $ref: '#/components/responses/ForbiddenError'
         500:
-          description: Internal Server Error.
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/responses/InternalServerError'
+          $ref: '#/components/responses/InternalServerError'
 
-    get:
-      operationId: Get settings
+    post:
+      operationId: Register integration
+      summary: Register a new cloud integration
       tags:
         - Management API
-      summary: Configure Azure IoT Hub settings.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Integration'
       responses:
         200:
-          description: Success.
+          description: OK. Returns the registered integration.
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Settings'
+                $ref: '#/components/schemas/Integration'
+        400:
+          $ref: '#/components/responses/InvalidRequestError'
         401:
-          description: Unauthorized.
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/responses/UnauthorizedError'
+          $ref: '#/components/responses/UnauthorizedError'
         403:
-          description: Forbidden.
+          $ref: '#/components/responses/ForbiddenError'
+        500:
+          $ref: '#/components/responses/InternalServerError'
+
+  /integrations/{id}:
+    delete:
+      operationId: Remove integration
+      summary: Unregister cloud integration
+      tags:
+        - Management API
+      parameters:
+        - name: id
+          in: path
+          description: Integration identifier.
+          required: true
+          schema:
+            type: string
+      responses:
+        204:
+          description: Integration unregistered successfully.
+        400:
+          $ref: '#/components/responses/InvalidRequestError'
+        401:
+          $ref: '#/components/responses/UnauthorizedError'
+        403:
+          $ref: '#/components/responses/ForbiddenError'
+        500:
+          $ref: '#/components/responses/InternalServerError'
+
+
+    # put: would be nice-to-have but not strictly necessary
+    #      we can live with delete + post for first iteration.
```

Comment:
Are you sure we can live without put? What if you need to change the connection string because of key-rotation?
This will generate a new integration ID.

Reply:
Added `PUT /integrations{id}/credentials`